flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/v2/PaimonWriter.java

```diff
@@ -186,4 +216,10 @@ public void close() throws Exception {
             compactExecutor.shutdownNow();
         }
     }
+
+    @Override
+    public List<PaimonWriterState> snapshotState(long checkpointId) throws IOException {
+        this.checkpointId = checkpointId;
```

Comment:
Seems `this.checkpointId` and `long checkpointId` have different meanings, and identical name might be confusing. Maybe we can rename field to `lastCheckpointId`, just like what `KafkaWriter` does?